rclcpp/test/rclcpp/test_generic_pubsub.cpp

```diff
@@ -263,3 +263,51 @@ TEST_F(RclcppGenericNodeFixture, generic_publisher_uses_qos)
   // It normally takes < 20ms, 5s chosen as "a very long time"
   ASSERT_TRUE(wait_for(connected, 5s));
 }
+
+TEST_F(RclcppGenericNodeFixture, generic_subscription_different_callbacks)
+{
+  // If the GenericSubscription does not use the provided QoS profile,
+  // its request will be incompatible with the Publisher's offer and no messages will be passed.
```

Comment:
I think this comment is more relevant to the test from which it was copied. I would remove it here. 
```suggestion
```